src/ucp/core/ucp_request.c

```diff
@@ -344,3 +352,52 @@ ucp_request_send_start(ucp_request_t *req, ssize_t max_short,
 
     return UCS_ERR_NO_PROGRESS;
 }
+
+int ucp_request_rndv_reg(ucp_request_t *req)
+{
+    ucp_ep_t        *ep      = req->send.ep;
+    ucp_dt_state_t  *state   = &req->send.state.dt;
+    int              cnt     = 0;
+    ucs_status_t     status;
+    int              i;
+    ucp_lane_index_t lane;
+
+    ucs_assert(UCP_DT_IS_CONTIG(req->send.datatype));
+
+    ucp_dt_clear_rndv_lanes(state);
```

Comment:
don't have to clear all lanes - can set memh to NULL as an 'else' clause in line 378